pkg/controller/account/iam.go

```diff
@@ -434,10 +436,14 @@ func deleteIAMUser(reqLogger logr.Logger, awsClient awsclient.Client, user *iam.
 	return nil
 }
 
+var (
+	listIAMUsers = awsclient.ListIAMUsers
+)
+
 func deleteIAMUsers(reqLogger logr.Logger, awsClient awsclient.Client, accountCR *awsv1alpha1.Account) error {
 	reqLogger.Info("Cleaning up IAM users")
 
-	users, err := awsclient.ListIAMUsers(reqLogger, awsClient)
+	users, err := listIAMUsers(reqLogger, awsClient)
```

Comment:
In my opinion
```
users, err := awsclient.ListIAMUsers(reqLogger, awsClient)
```
is much better, cleaner and easier to understand than:
```
var listIAMUsers = awsclient.ListIAMUsers
users, err := listIAMUsers(reqLogger, awsClient)
```
I prefer leaving it the way it was unless there is a reason we are changing it that I don't understand